fbw-common/src/systems/shared/src/publishers/ArincBus/IrBusPublisher.ts

```diff
@@ -0,0 +1,326 @@
+  ir_body_longitudinal_acc: number;
+  /** The lateral (left/right) acceleration of the aircraft in g's. Raw ARINC Word. */
+  ir_body_lateral_acc: number;
+  /** The normal acceleration (load factor) of the aircraft in g's. Raw ARINC Word. */
+  ir_body_normal_acc: number;
+  /** The heading rate (ψ^dot) of the aircraft in degrees/sec. Raw ARINC Word. */
+  ir_heading_rate: number;
+  /** The pitch rate (θ^dot) of the aircraft in degrees/sec. Raw ARINC Word. */
+  ir_pitch_att_rate: number;
+  /** The roll rate (φ^dot) of the aircraft in degrees/sec. Raw ARINC Word. */
+  ir_roll_att_rate: number;
+  /**
+   * Indicates state of the IR. Raw ARINC Word.
+   * Bit(s)   | Meaning
+   * -------- | --------------------------
+   *        0 | ALIGNMENT_NOT_READY
+   *        1 | REV_ATT_MODE
+   *        2 | NAV_MODE
+   *        3 | VALID_SET_HEADING
+   *        4 | ATTITUDE_INVALID
+   *        5 | DC_FAIL
+   *        6 | ON_DC
+   *        7 | ADR_FAULT
+   *        8 | IR_FAULT
+   *        9 | DC_FAIL_ON_DC
+   *       10 | ALIGN_FAULT
+   *       11 | NO_IRS_INITIAL
+   *       12 | EXCESS_MOTION_ERROR
+   *       13 | ADR_IR_FAULT
+   *       14 | EXTREME_LATITUDE
+   * 15,16,17 | ALIGN_7_10_MINUTES
+   *    16,17 | ALIGN_6_MINUTES
+   *    15,17 | ALIGN_5_MINUTES
+   *       17 | ALIGN_4_MINUTES
+   *    15,16 | ALIGN_3_MINUTES
+   *       16 | ALIGN_2_MINUTES
+   *       15 | ALIGN_1_MINUTES
+   *       18 | COMPUTED_LATITUDE_MISCOMPARE
+   */
+  ir_maint_word: number;
+}
+
+type IndexedTopics = keyof IrBusBaseEvents;
+
+type IrIndexedEventType<T extends string> = `${T}_${1 | 2 | 3}`;
+
+type IrBusIndexedEvents = {
+  [P in keyof Pick<IrBusBaseEvents, IndexedTopics> as IrIndexedEventType<P>]: IrBusBaseEvents[P];
+};
+
+interface IrBusPublisherEvents extends IrBusBaseEvents, IrBusIndexedEvents {}
+
+/**
+ * Events for A32NX IR output bus local vars.
+ */
+export interface IrBusEvents extends Omit<IrBusBaseEvents, IndexedTopics>, IrBusIndexedEvents {}
+
+/**
+ * Publisher for A32NX IR output bus local vars.
+ */
+export class IrBusPublisher extends SimVarPublisher<IrBusPublisherEvents> {
+  /**
+   * Create a publisher.
+   * @param bus The EventBus to publish to
+   * @param pacer An optional pacer to use to control the rate of publishing
+   */
+  public constructor(bus: EventBus, pacer?: PublishPacer<IrBusPublisherEvents>) {
+    const simvars = new Map<keyof IrBusPublisherEvents, SimVarPublisherEntry<any>>([
+      [
+        'ir_pitch',
+        {
+          name: 'L:A32NX_ADIRS_IR_#index#_PITCH',
+          type: SimVarValueType.Enum,
+          indexed: true,
+        },
+      ],
+      [
+        'ir_roll',
+        {
+          name: 'L:A32NX_ADIRS_IR_#index#_ROLL',
+          type: SimVarValueType.Enum,
+          indexed: true,
+        },
+      ],
+      [
+        'ir_heading',
+        {
+          name: 'L:A32NX_ADIRS_IR_#index#_HEADING',
+          type: SimVarValueType.Enum,
```

Comment:
Any reason all of those are type "Enum"?

Reply:
We want integers only, no decimals.